lib/utils/tls.go

```diff
@@ -54,6 +62,74 @@ type CertificateName struct {
 	OrganizationalUnit []string `json:"org_unit"`
 }
 
+// TLSCredentials keeps the typical 3 components of a proper HTTPS configuration
+type TLSCredentials struct {
+	// PublicKey in PEM format
+	PublicKey []byte
+	// PrivateKey in PEM format
+	PrivateKey []byte
+	Cert       []byte
+}
+
+// GenerateSelfSignedCert generates a self signed certificate that
+// is valid for given domain names and ips, returns PEM-encoded bytes with key and cert
+func GenerateSelfSignedCert(hostNames []string) (*TLSCredentials, error) {
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	notBefore := time.Now()
+	notAfter := notBefore.Add(time.Hour * 24 * 825) // 825 days or fewer is the required validity period for MacOS
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	entity := pkix.Name{
+		CommonName:   "localhost",
+		Country:      []string{"US"},
+		Organization: []string{"localhost"},
+	}
+
+	template := x509.Certificate{
+		SerialNumber:          serialNumber,
+		Issuer:                entity,
+		Subject:               entity,
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
```

Comment:
Add a comment here to indicate this is a part of the macOS cert requirements, same as you did above for the validity duration?